File: Frequency of the most frequent element.cpp
class Solution {
public:
    int maxFrequency(vector<int>& nums, long k) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int j = 0;
        int ans = 0;
        for(int i = 0; i < n; i++) {
            k =k+ nums[i];

            while(k < (long) nums[i] * (i - j + 1)) {
                k =k- nums[j];
                j++;
            }
            ans = max(ans, i - j + 1);
        }
        return ans;
    }
};

File: first-phase/week2/frequency-of-the-most-frequent-element.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        freq = 0
        i = 0
        j = 1
        nums.sort(reverse = True)
        if len(nums) == 1:
            return 1
        while j < len(nums):
            while nums[i] - nums[j] > k:
                i+=1
                k += (nums[i-1] - nums[i])*(j-i)
            freq=max(freq,j-i+1)
            k -= (nums[i] - nums[j])
            j+=1
        return freq
        
